ambry-api/src/main/java/com.github.ambry/rest/RestUtils.java

```diff
@@ -456,7 +476,12 @@ public static String getOperationOrBlobIdFromUri(RestRequest restRequest, RestUt
       // "- 1" removes the "slash" that precedes the sub-resource.
       endIndex = endIndex - subResource.name().length() - 1;
     }
-    return path.substring(startIndex, endIndex);
+    String operationOrBlobId = path.substring(startIndex, endIndex);
+    if ((operationOrBlobId.isEmpty() || operationOrBlobId.equals("/")) && restRequest.getArgs()
+        .containsKey(Headers.BLOB_ID)) {
+      operationOrBlobId = restRequest.getArgs().get(Headers.BLOB_ID).toString();
```

Comment:
It's caught by the business logic. For example, if you do a GET request with URL http://localhost:1174/ today (no blob ID) and you would get `BadRequest`. So adding this piece of logic makes no difference since this function could have returned an empty string even before this change. Does that answer your question?